.github/workflows/preview.yml

```diff
@@ -0,0 +1,42 @@
+name: Preview
+on:
+  pull_request:
+    branches: 
+      - master
+      - develop
+      - 'release/**'
```

Comment:
A bit of a counter to what you're trying to do, but it is probably best to not listen to all activity on these branches. Instead, kicking off the workflow when a `push` is made to the `master` or `develop` branch. As more developers are opening up PRs against `develop`, we might quickly reach our storage limits in S3.

Reply:
@jmelberg-okta thanks for calling this out. This is something I've thought about as well and talked about with @edburyenegren-okta 

The way it (currently) works is that it only deploys a preview if you push to an open PR. We discussed adding a condition, choosing to deploy _only if the PR *is not* in draft state_ do you think this would help or solve the issue?

Reply:
It helps, but the more we store the more expensive it gets. A couple of questions probably better suited for the labs team:

1. Can we cleanup stale versions of the docs site after `n` days?
2. As our doc site continues to grow in size, what upload limit does GH workflows/aws CLI support?
3. Can we limit GH workflow execution to only certain teams?

Reply:
If one of these options is necessary, I'd prefer a time-boxed, no-update or on-merge-to-master cleanup over anything gating on teams or contributors.

Archival is neat, but not necessary and definitely not useful after a certain point - especially since the git history allows a time machine if needed. Working there to save space seems like the smallest/healthiest compromise.

Reply:
I can answer these questions

> 1. Can we cleanup stale versions of the docs site after n days?

Yes. We can set up [object expiration](https://aws.amazon.com/blogs/aws/amazon-s3-object-expiration/#:~:text=S3's%20new%20Object%20Expiration%20function,from%20the%20AWS%20Management%20Console.) rules on the bucket.

> 2. As our doc site continues to grow in size, what upload limit does GH workflows/aws CLI support?

Neither workflows nor `aws` cli have an upload limit that I am aware of. Further, aws-cli recommends that you upload large files via the CLI. Our monorepo should never produce massive builds, even as we add more and more packages to it.

Further, GitHub Actions has a [6-hour execution time-out](https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits). 

If we cant upload our preview build(s) within a few minutes (_at most_), we have a problem.

> 3. Can we limit GH workflow execution to only certain teams?

We can, but I firmly believe anyone who pushes code to this repository should be able to benefit from a preview environment. If the concern here is space, I think that can easily be mitigated by the aforementioned [object expiration](https://aws.amazon.com/blogs/aws/amazon-s3-object-expiration/#:~:text=S3's%20new%20Object%20Expiration%20function,from%20the%20AWS%20Management%20Console.) rules.

Reply:
Best practice is to limit deploy-like actions to only contributors, and (moving forward) we're going to likely attach more business logic to these actions. Can we restrict access now, then open it up if needed?

Reply:
If we do this, I would like for it to be anyone within the okta org. Is that cool?

If so ill do some research into how to make this happen.

Reply:
I think we're probably good with whoever has write access to the repo (DS/Eng-push teams). Since we verified forked PRs do not trigger workflows - I don't think there is action needed on our end.

Reply:
Makes sense, I will mark this as resolved.